checkov/cloudformation/graph_builder/variable_rendering/renderer.py

```diff
@@ -0,0 +1,206 @@
+
+        if all(isinstance(element, str) for element in value) and \
+                condition_name == dest_vertex_attributes.get(CustomAttributes.BLOCK_NAME) and \
+                dest_vertex_attributes.get(CustomAttributes.BLOCK_TYPE) == BlockType.CONDITIONS:
+            evaluated_value = self._evaluate_condition(value)
+            evaluated_value = str(evaluated_value) if evaluated_value else None
+        return evaluated_value
+
+    def _evaluate_condition(self, value: List[str]) -> Optional[bool]:
+        # value = [condition_name, value_if_true, value_if_false]
+        return None
+
+    def update_evaluated_value(
+            self,
+            changed_attribute_key: str,
+            changed_attribute_value: Union[str, List[str]],
+            vertex: int,
+            change_origin_id: int,
+            attribute_at_dest: Optional[Union[str, List[str]]] = None,
+    ) -> None:
+        """
+        The function updates the value of changed_attribute_key with changed_attribute_value for vertex
+        """
+        self.local_graph.update_vertex_attribute(
+            vertex, changed_attribute_key, changed_attribute_value, change_origin_id, attribute_at_dest
+        )
+
+    @staticmethod
+    def find_path_from_referenced_vertices(
+            referenced_vertices: List[TerraformVertexReference], vertex_attributes: Dict[str, Any]
+    ) -> Tuple[List[str], str]:
+        """
+        :param referenced_vertices: an array of VertexReference
+        :param vertex_attributes: attributes to search
+        :return attribute_path: [] if referenced_vertices does not contain vertex_attributes,
+                                else the path to the searched attribute
+        :return origin_value
+        """
+        for vertex_reference in referenced_vertices:
+            block_type = vertex_reference.block_type
+            attribute_path = vertex_reference.sub_parts
+            if vertex_attributes[CustomAttributes.BLOCK_TYPE] == block_type:
+                for i in range(len(attribute_path)):
+                    name = ".".join(attribute_path[: i + 1])
+                    if vertex_attributes[CustomAttributes.BLOCK_NAME] == name:
+                        return attribute_path, vertex_reference.origin_value
+        return [], ""
+
+    def _extract_vertices_block_name_map(self) -> Dict[str, Dict[str, List[int]]]:
+        vertices_block_name_map = deepcopy(self.local_graph.vertices_block_name_map)
+        resources_blocks_name_map = vertices_block_name_map[BlockType.RESOURCE]
+
+        updated_resources_blocks_name_map = {}  # used in order to prevent RuntimeError: dictionary changed size during iteration
+        for resource_name, blocks_list in resources_blocks_name_map.items():
+            shortened_resource_name = resource_name.split('.')[-1]  # Trims AWS::X::Y.ResourceName and leaves us with ResoruceName
+            updated_resources_blocks_name_map[shortened_resource_name] = blocks_list
```

Comment:
do you mean this is not possible to write as a dict comprehension, due to a possible runtime error?

Reply:
Forgot to remove the comment while working on it. TBH I don't want to use dict comprehension here because I prefer it to be readable with the comment in the loop